### ISSUE TITLE
Tabindex attribute of Vaadin button gets out of sync

### ISSUE THREAD

**Issue by eforest** [vaadin/vaadin-flow-components#626]
Tabindex attribute of Vaadin button gets out of sync.
In the end the button is out of tab order and cannot be accessed by keyboard any more.

To reproduce click through this demo. At step 4 you can't access save button by keyboard any more.
Used Vaadin 14.4.0.

```
@Route("button")
@RouteAlias("")
public class DemoDialog extends Div {

    private static final long serialVersionUID = 1L;

    public DemoDialog() {
        getStyle().set("margin", "1rem");

        VerticalLayout buttons = new VerticalLayout();

        // button is focusable and can be accessed by tab initially
        Button save = new Button("save");
        save.setTabIndex(10);

        // disabling also removes tabindex attribute
        Button b1 = new Button("1. disable save");
        b1.addClickListener(e -> save.setEnabled(false));

        // now it's gone
        Button b2 = new Button("2. remove save from view");
        b2.addClickListener(e -> buttons.remove(save));

        // added again, still disabled, but java comp and js comp are not in sync any more
        Button b3 = new Button("3. add save to view");
        b3.addClickListener(e -> buttons.add(save));

        // enabling doesn't set tabindex property
        // even if i set it explicitly again that doesn't change
        // so now save button is out of tab order and cannot be accessed by keyboard any more
        Button b4 = new Button("4. enable save");
        b4.addClickListener(e -> save.setEnabled(true));

        // workaround is to set another value for tabindex
        Button b5 = new Button("5. get save focusable again");
        b5.addClickListener(e -> save.setTabIndex(5));

        buttons.add(b1, b2, b3, b4, b5, save);

        add(buttons);
    }

}

```
